runtime/vam/expr/compare.go

```diff
@@ -0,0 +1,960 @@
+package expr
+
+import (
+	"fmt"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/runtime/sam/expr/coerce"
+	"github.com/brimdata/zed/vector"
+)
+
+type Compare struct {
+	zctx *zed.Context
+	op   string
+	lhs  Evaluator
+	rhs  Evaluator
+}
+
+func NewCompare(zctx *zed.Context, lhs, rhs Evaluator, operator string) *Compare {
+	return &Compare{zctx, operator, lhs, rhs}
+}
+
+func (c *Compare) Eval(val vector.Any) vector.Any {
+	l := vector.Under(c.lhs.Eval(val))
+	r := vector.Under(c.rhs.Eval(val))
+	lhs, rhs, _ := coerceVals(c.zctx, l, r)
```

Comment:
Forgive the naive comment, but this is going to panic if val is a Variant (since calling Type(), as coerceVals does, on vector.Variant panics). Would the idea in a future update have this recursively call Compare if Variant is encountered here?

Reply:
Nvm, I suppose, I see the comment `//XXX unions and variants and single-value-with-error variant`.